test/modules/post/test/smb.rb

```diff
@@ -0,0 +1,203 @@
+require 'rex/post/meterpreter/extensions/stdapi/command_ids'
+require 'rex'
+require 'fileutils'
+
+lib = File.join(Msf::Config.install_root, "test", "lib")
+$LOAD_PATH.push(lib) unless $LOAD_PATH.include?(lib)
+require 'module_test'
+
+class MetasploitModule < Msf::Post
+
+  include Msf::ModuleTest::PostTest
+  include Msf::ModuleTest::PostTestFileSystem
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Testing SMB sessions work',
+        'Description' => %q{ This module will test the SMB sessions work },
+        'License' => MSF_LICENSE,
+        'Author' => [ 'sjanusz-r7'],
+        'Platform' => all_platforms,
+        'SessionTypes' => [ 'smb' ]
+      )
+    )
+  end
+
+  def setup
+    super
+  end
+
+  def cleanup
+    super
+  end
+
+  def test_console_help
+    it "should support the help command" do
+      stdout = with_mocked_console(session) { |console| console.run_single("help") }
+      ret = true
+      ret &&= stdout.buf.include?('Core Commands')
+      ret &&= stdout.buf.include?('Shares Commands')
+      ret &&= stdout.buf.include?('Local File System Commands')
+      ret
+    end
+  end
+
+  def test_file_upload_and_download
+    filename = Rex::Text.rand_text_alphanumeric(8..16)
+    file_path = File.expand_path "/tmp/#{filename}"
```

Comment:
iirc there's already helpers for creating local test files that are available in the other test modules? 👀